test/dynamo/test_sets.py

```diff
@@ -0,0 +1,225 @@
+# Owner(s): ["module: dynamo"]
+
+# TODO: move set tests from test_functions.py/test_misc.py to this file
+
+import math
+import unittest
+
+import torch
+import torch._dynamo.test_case
+from torch._dynamo.testing import CompileCounter
+from torch.testing._internal.common_utils import munge_exc
+from torch.testing._internal.logging_utils import LoggingTestCase, make_logging_test
+
+
+class TestSetGuards(LoggingTestCase):
+    def test_set_with_tensor(self):
+        s = {
+            torch._C._set_grad_enabled,
+            torch.randn(2),
+            torch.amp._exit_autocast,
+        }
+        cnts = CompileCounter()
+
+        @torch.compile(backend=cnts, fullgraph=True)
+        def fn(x, s):
+            if torch.amp._exit_autocast in s:
+                return x.sin()
+            return x.cos()
+
+        x = torch.randn(2)
+        y = fn(x, s)
+        self.assertEqual(y, x.sin())
+        self.assertEqual(cnts.frame_count, 1)
+
+        s.remove(torch.amp._exit_autocast)
+        s.add(torch._C._set_fwd_grad_enabled)
+        y = fn(x, s)
+        self.assertEqual(y, x.cos())
+        self.assertEqual(cnts.frame_count, 2)
+
```

Comment:
@guilhermeleobas I don't have a good explanation for the following:
<img width="499" alt="image" src="https://github.com/user-attachments/assets/010650f3-5abf-4fed-9757-88a9c8876cfb" />

Reply:
nvm It turns out IPython has a pretty-printer that has its own ordering for sets.